editor/src/java/com/defold/editor/pipeline/TextureSetGenerator.java

```diff
@@ -370,21 +372,18 @@ public static TextureSetResult generate(List<BufferedImage> images, List<Integer
 
         List<Rect> imageRects = rectanglesFromImages(images, paths);
 
-        int usesSpriteTrimming = 0;
-        for (Integer hullSize : imageHullSizes) {
-            usesSpriteTrimming += hullSize;
-        }
-
         // if all sizes are 0, we still need to generate hull (or rect) data
         // since it will still be part of the new code path if there is another atlas with trimming enabled
         List<SpriteGeometry> imageHulls = new ArrayList<SpriteGeometry>();
+        int use_geometries = 0;
         for (int i = 0; i < images.size(); ++i) {
             BufferedImage image = images.get(i);
+            use_geometries |= imageHullSizes.get(i) > 0 ? 1 : 0;
             imageHulls.add(buildConvexHull(image, imageHullSizes.get(i)));
         }
 
         // The layout step will expand the rect, and possibly rotate them
-        TextureSetResult result = calculateLayout(imageRects, imageHulls, iterator,
+        TextureSetResult result = calculateLayout(imageRects, imageHulls, use_geometries, iterator,
```

Comment:
This is indeed interesting.
While I was implementing this feature, it took a lot of time and effort to investigate the differences between this file and its counterpart (in bob). Eventually, I made it at least be identical to the other one, as stated at the top of the file:
> // The code below must remain identical to the implementation in the editor!
// ./editor/src/java/com/defold/editor/pipeline/TextureSetGenerator.java

But clearly, I failed to maintain my promise, thus resulting in this bug :(

Reply:
Yes, our goal is to remove these duplicate files, by calling them from bob.jar instead, since that's always integrated into the editor.